include/onnxruntime/core/session/onnxruntime_cxx_inline.h

```diff
@@ -1555,7 +1555,7 @@ inline const OrtMemoryInfo* CustomOpApi::GetTensorMemoryInfo(_In_ const OrtValue
 
 template <typename T>
 inline const T* CustomOpApi::GetTensorData(_Inout_ const OrtValue* value) {
-  return GetTensorData<T>(value);
+  return GetTensorMutableData<T>(const_cast<OrtValue*>(value));
```

Comment:
>GetTensorMutableData<T>(const_cast<OrtValue*>(value)); [](http://example.com/codeflow?start=9&length=54)

Came to think of it. It would be better to direct it straight to C API. Then it would not recurse.